core/src/main/scala/kafka/consumer/BaseConsumer.scala

```diff
@@ -59,6 +59,9 @@ class NewShinyConsumer(topic: Option[String], partitionId: Option[Int], offset:
   import org.apache.kafka.clients.consumer.KafkaConsumer
 
   val consumer = new KafkaConsumer[Array[Byte], Array[Byte]](consumerProps)
+  val consumedOffsets = collection.mutable.Map[TopicPartition, Long]()
```

Comment:
There's really no reason for this to be a field. We can make it a local variable in `seekRealPositionsBeforeExit`.

Reply:
`consumedOffsets` will be filled in and updated in `updateOffsetForPartition`. Does it make sense?